pkg/kubectl/cmd/util/helpers.go

```diff
@@ -122,6 +122,11 @@ func checkErrWithPrefix(prefix string, err error) {
 // checkErr formats a given error as a string and calls the passed handleErr
 // func with that string and an kubectl exit code.
 func checkErr(prefix string, err error, handleErr func(string, int)) {
+	// unwrap aggregates of 1
+	if agg, ok := err.(utilerrors.Aggregate); ok && len(agg.Errors()) == 1 {
```

Comment:
NewAggregate gives you an aggregate - if we only conditionally gave you an aggregate, clients could silently be broken.  Instead, we have to add a new method and migrate folks over to it.